app/controllers/entities_controller.rb

```diff
@@ -0,0 +1,37 @@
+class EntitiesController < ApplicationController
+  before_action :authenticate_user!
+  def index
+    @group = current_user.groups.find(params[:group_id])
+    @entities = @group.entities
+  end
+
+  def new
+    @group = current_user.groups.find(params[:group_id])
+    @entity = @group.entities.new
+  end
+
+  def create
+    @group = current_user.groups.find(params[:group_id])
+    @entity = current_user.entities.create(entity_params)
+    puts @entity
+    if @entity.save
+      @group_entity = @entity.group_entities.create(group_id: @group.id, entity_id: @entity.id)
+      if @group_entity.save
+        flash[:notice] = 'Transaction created successfully'
+        redirect_to group_entities_path(@group)
+      else
+        flash.now[:alert] = 'An error occuerd , Transaction category creation failed'
+        render action: 'new'
+      end
+    else
+      flash.now[:alert] = 'An error occuerd , Transaction creation failed'
+      render action: 'new'
+    end
+  end
```

Comment:
- [ ] it would be better if you add an inputs of category at least one when creating a transaction in order to match the requirements 
<img width="1016" alt="Screen Shot 2022-10-14 at 18 28 20" src="https://user-images.githubusercontent.com/61790587/195906697-bce33421-4eee-4866-973e-f87632042521.png">